packages/env-options/src/env-options.js

```diff
@@ -0,0 +1,98 @@
+// @ts-check
+
+// `@endo/env-options` needs to be imported quite early, and so should
+// avoid importing from ses or anything that depends on ses.
+
+// /////////////////////////////////////////////////////////////////////////////
+// Prelude of cheap good - enough imitations of things we'd use or
+// do differently if we could depend on ses
+
+const { freeze } = Object;
+const { apply } = Reflect;
+
+// Should be equivalent to the one in ses' commons.js even though it
+// uses the other technique.
+const uncurryThis =
+  fn =>
+  (receiver, ...args) =>
+    apply(fn, receiver, args);
+const arrayFlatMap = uncurryThis(Array.prototype.flatMap);
+const arrayPush = uncurryThis(Array.prototype.push);
+const arrayJoin = uncurryThis(Array.prototype.join);
+
+const q = JSON.stringify;
+
+const Fail = (template, ...args) => {
+  const msg = arrayFlatMap(args, (arg, i) => [template[i], arg]);
+  arrayPush(msg, template[args.length]);
+  throw Error(arrayJoin(msg, ''));
+};
+
+// end prelude
+// /////////////////////////////////////////////////////////////////////////////
+
+/**
+ * `makeEnvironmentCaptor` provides a mechanism for getting environment
+ * variables, if they are needed, and a way to catalog the names of all
+ * the environment variables that were captured.
+ *
+ * @param {object} aGlobal
+ */
+export const makeEnvironmentCaptor = aGlobal => {
+  const capturedEnvironmentOptionNames = [];
+
+  /**
+   * Gets an environment option by name and returns the option value or the
+   * given default.
+   *
+   * @param {string} optionName
+   * @param {string} defaultSetting
+   * @returns {string}
+   */
+  const getEnvironmentOption = (optionName, defaultSetting) => {
+    // eslint-disable-next-line @endo/no-polymorphic-call
+    typeof optionName === 'string' ||
+      Fail`Environment option name ${q(optionName)} must be a string.`;
+    // eslint-disable-next-line @endo/no-polymorphic-call
+    typeof defaultSetting === 'string' ||
+      Fail`Environment option default setting ${q(
+        defaultSetting,
+      )} must be a string.`;
+
+    /** @type {string} */
+    let setting = defaultSetting;
+    const globalProcess = aGlobal.process;
+    if (globalProcess && typeof globalProcess === 'object') {
+      const globalEnv = globalProcess.env;
+      if (globalEnv && typeof globalEnv === 'object') {
+        if (optionName in globalEnv) {
+          arrayPush(capturedEnvironmentOptionNames, optionName);
+          const optionValue = globalEnv[optionName];
```

Comment:
Performing this lookup against `globalEnv` in every call creates the possibility of results changing—should we instead memoize them by `optionName` (e.g., replacing the `capturedEnvironmentOptionNames` array into a `capturedEnvironmentOptions` map)?

Reply:
Options setting code, to be tolerant of the absence of a `globalThis.process.env` object, currently looks like
```js
const env = (globalThis.process || {}).env || {};
env.TRACK_TURNS = 'enabled';
```
This kinda defeats one of the purposes of this package, which is to raise the abstraction level.
In a later PR I intend to provide an API for setting options. That one will determine our stance on these temporal issues. In the meantime, the current code is a good starting point for what I expect to do.

Resolving.